shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/service/StudyCardProcessingService.java

```diff
@@ -14,108 +14,355 @@
+import org.shanoir.ng.studycard.model.Field;
 import org.shanoir.ng.studycard.model.Operation;
 import org.shanoir.ng.studycard.model.StudyCard;
+import org.shanoir.ng.studycard.model.StudyCardApply;
 import org.shanoir.ng.studycard.model.StudyCardAssignment;
 import org.shanoir.ng.studycard.model.StudyCardCondition;
+import org.shanoir.ng.studycard.model.StudyCardConditionValue;
 import org.shanoir.ng.studycard.model.StudyCardRule;
+import org.shanoir.ng.studycard.model.StudyCardRuleType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.amqp.AmqpException;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Service
 public class StudyCardProcessingService {
 	
 	private static final Logger LOG = LoggerFactory.getLogger(StudyCardProcessingService.class);
-
 	
+	@Autowired
+	private StudyCardService studyCardService;
+	
+	@Autowired
+	private SubjectStudyRepository subjectStudyRepository;
+	
+	@Autowired
+	private ExaminationService examinationService;
+	
+	@Autowired
+	private DatasetAcquisitionService datasetAcquisitionService;
+	
+	@Autowired
+	private WADODownloaderService downloader;
+	
+	@Autowired
+	private RabbitTemplate rabbitTemplate;
+	
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	/**
+	 * Application during import, when dicoms are present in tmp directory.
+	 * @param acquisition
+	 * @param studyCard
+	 * @param dicomAttributes
+	 */
 	public void applyStudyCard(DatasetAcquisition acquisition, StudyCard studyCard, Attributes dicomAttributes) {
-		LOG.debug("apply studycard n° " + studyCard.getId());
 		if (studyCard.getRules() != null) {
 			for (StudyCardRule rule : studyCard.getRules()) {
 				applyStudyCardRule(acquisition, rule, dicomAttributes);
 			}
 		}
 		acquisition.setStudyCard(studyCard);
 		acquisition.setStudyCardTimestamp(studyCard.getLastEditTimestamp());
+	}	
+
+	/**
+	 * Re-application on using web GUI and list of dataset acquisitions.
+	 * @param studyCardApplyObject
+	 */
+	public void applyStudyCard(StudyCardApply studyCardApplyObject) {
+		StudyCard studyCard = studyCardService.findById(studyCardApplyObject.getStudyCardId());
+		LOG.debug("re-apply studycard n° " + studyCard.getId());
+		List<DatasetAcquisition> acquisitions = datasetAcquisitionService.findById(studyCardApplyObject.getDatasetAcquisitionIds());
+		boolean changeInAtLeastOneAcquisition = false;
+		for (DatasetAcquisition acquisition : acquisitions) {
+			if (CollectionUtils.isNotEmpty(acquisition.getDatasets())) {
+				Dataset refDataset = acquisition.getDatasets().get(0);
+				Attributes dicomAttributes = getDicomAttributesForDataset(refDataset);
+				if (studyCard.getRules() != null) {
```

Comment:
Test for studyCard rules emptyness before loading all dicom attributes ?

Reply:
Yes, well seen, done.